src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -1873,3 +1871,146 @@ def match_attachment(request):
             document_create_attachment_body["vertrouwelijkheidaanduiding"], "geheim"
         )
         self.assertEqual(document_create_attachment_body["titel"], "A Custom Title")
+
+    @override_settings(ESCAPE_REGISTRATION_OUTPUT=True)
+    def test_submission_with_objects_api_escapes_html(self, m):
```

Comment:
I think I would simplify this test a bit with an explicit content JSON template. This also gives the opportunity to test that manual context usage is escaped (right now the test mostly asserts that the `{% json_summary %}` output is escaped).

To do that, I would use a custom `content_json` template, which you can pass via `objects_form_options`:

```py
content_template = """
{
  "summary": {% json_summary %},
  "manual_variable": "{{ voornaam }}"
}
"""
objects_form_options = {
    ...,
    "content_json": content_template,
}
```

You can then make the assertion on the request body for the part that's relevant:

```py
posted_record_data = object_create_body["record"]["data"]
expected_record_data = {
    "summary": {
      step_slug: {
        "voornaam": "&lt;script&gt;alert();&lt;/script&gt;",
      },
    },
    "manual_variable": "&lt;script&gt;alert();&lt;/script&gt;"
}
```